src/sentry/issues/search.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import annotations
+
+import functools
+from copy import deepcopy
+from typing import Any, Callable, Iterable, Mapping, Optional, Protocol, Sequence, Set, TypedDict
+
+from sentry import features
+from sentry.api.event_search import SearchFilter, SearchKey, SearchValue
+from sentry.models import Environment, Organization
+from sentry.search.events.filter import convert_search_filter_to_snuba_query
+from sentry.types.issues import GroupCategory
+from sentry.utils import snuba
+from sentry.utils.snuba import SnubaQueryParams, raw_query_params
+
+
+class SearchQueryPartial(Protocol):
+    def __call__(
+        self, conditions: Sequence[Any], aggregations: Sequence[Any], condition_resolver: Any
+    ) -> Mapping[str, Any]:
+        ...
+
+
+GroupSearchStrategy = Callable[
+    [
+        Set[GroupCategory],
+        Sequence[Any],
+        SearchQueryPartial,
+        int,
+        Sequence[int],
+        Optional[Sequence[Environment]],
+        Sequence[Any],
+    ],
+    Optional[SnubaQueryParams],
+]
+
+
+class MergeableRow(TypedDict, total=False):
+    group_id: int
+
+
+class GroupSearchParamsMapper:
+    """
+    Based on the search query used, we may need to 'split', map, and query multiple sources to satisfy the search.
+    This class maps the GroupCategories being searched to the appropriate function(s) to be used for generating the
+    parameters to then be used for bulk-querying snuba in parallel.
+    """
+
+    @staticmethod
+    def search_strategies_for(
+        group_categories: Iterable[GroupCategory],
+    ) -> Sequence[GroupSearchStrategy]:
+        return (
+            list(GroupSearchParamsMapper.search_strategies().values())
+            if not group_categories
+            else [GroupSearchParamsMapper.search_strategies()[gc] for gc in group_categories]
+        )
+
+    @staticmethod
+    def search_strategies() -> Mapping[GroupCategory, GroupSearchStrategy]:
+        return {
+            GroupCategory.ERROR: GroupSearchParamsMapper._query_params_for_error,
+            GroupCategory.PERFORMANCE: GroupSearchParamsMapper._query_params_for_perf,
+        }
```

Comment:
Nit: Might be worth defining this directly on the class so that we don't recreate every time we call the method.

Reply:
On this note, what do you think about this class in general? I'm thinking I should unfurl this into the file and get rid of the class altogether. 

Generally I like to bundle things into classes instead of sticking it into a file with a bunch of functions but that may not be pythonic?

Be interested to hear your thoughts.

Reply:
It's really just personal preference, but I don't like classes that are a collections of static methods - they may as well just be part of a module together. I don't feel strongly though, so if you want to keep it as a class it's nbd